highfidelity/hifi
# AnimRandomSwitch class for random idle implementation

The associated cases are here: 
1) https://highfidelity.manuscript.com/f/cases/22179/Add-random-switch-mode-for-long-idles
2) https://highfidelity.manuscript.com/f/cases/22181/
3) https://highfidelity.manuscript.com/f/cases/22180/

### Test Plan:

For 22179
1) open interface in desktop mode, 3rd person 
2) enable the developer-> "animation stats" menu
3) watch the middle column which displays the animation transitions and what is playing currently
4) you should see idletalk: "some motion" , some motion should change between idleTalk1,idleTalk2 and idleTalk3  every 5-10 seconds. sometimes it will choose the same motion again and will not change.
5) you should see the top row change between idleStand: masterIdle, and fidget:movement and aternateIdle:altIdle1 or altIdle2 
6) overall you should see the avatar occasionally sway in the hips when standing, occasionally adjust the feet. and occasionally look around. FYI: you must disable ik by de-selecting developer->avatar->enable inverse kinematics to see the head movement.
7) you should also see the arms go up when you talk into your un-muted mic.

For 22181
1) open interface in desktop in 3rd person
2) make sure that your mic is not muted
3) when you start talking into the mic you should see the talking animation blend into the already running animation. At first this takes a few seconds to get the hands up but most of the time when you talk the hands will go up in front of the avatar.

For 22180
1) open interface in desktop mode in 3rd person
2) enable developer->"show animation stats" menu
3) press the forward arrow key to move forward and let the avatar settle, do this repeatedly
4) expected result: in the animation stats you should see the masteridle change between masteridle1, masteridle2,m and masteridle3 this won't change every time because it is possible for the same idle to be chosen randomly twice in a row.
